src/prefect/tasks/monte_carlo/client.py

```diff
@@ -0,0 +1,205 @@
+import pendulum
+import requests
+from typing import Any, Dict, List
+
+from prefect.utilities.logging import get_logger
+
+
+class MonteCarloClient:
+    def __init__(
+        self,
+        api_key_id: str,
+        api_token: str,
+    ) -> None:
+        self.api_key_id = api_key_id
+        self.api_token = api_token
+        self.logger = get_logger()
+        self._api_url = "https://api.getmontecarlo.com/graphql"
+
+    def _send_graphql_request(
+        self, query: str, variables: dict = None
+    ) -> Dict[str, Any]:
+        response = requests.post(
+            url=self._api_url,
+            json=dict(query=query, variables=variables),
+            headers={
+                "x-mcd-id": self.api_key_id,
+                "x-mcd-token": self.api_token,
+                "Content-Type": "application/json",
+            },
+        )
+        response.raise_for_status()
+        response = response.json()
+        self.logger.info("Response: %s", response)
```

Comment:
Do we want info level logs for every response? Seems more like a debug level or not worth including for users. There's also no request information here so there's no context for this response.

Reply:
Good point, I will change it to debug. The response provides the query/mutation information after the `data` dictionary key, e.g. this query:

```python
query="""
      mutation($node_name: String!, $object_id: String!, $object_type: String!,
      $resource_name: String! ) {
        createOrUpdateLineageNode(
          name: $node_name,
          objectId: $object_id,
          objectType: $object_type,
          resourceName: $resource_name,
        ){
          node{
            nodeId
            mcon
          }
        }
      }
      """

```

generates a response: 

```python
{'data': {'createOrUpdateLineageNode': {'node': {'mcon': 'MCON++15d3444b-f063-4de1-9b31-104f44223a79++3cc41ba1-132c-43cd-b6ea-ea68fe4a2906++table++new_demo_source'}}}}
```

but it doesn’t return info about which variables where passed to the query/mutation. I will include both query and variables:

```python
self.logger.debug("Response: %s for request %s with variables %s", response, query, variables)
```

LMK if I should change anything in this line.